### PR TITLE
PNDA-4513: Flink Metrics - Configure Graphite Metrics reporter

### PR DESCRIPTION
**Pull request by GaneshManal**
# Problem Statement:
PNDA-4513: Flink Metrics - Configure Graphite Metrics reporter

# Changes:
Below configurations are required -
- Update flink file ( .yaml ) for configuring graphite as reporter for monitoring data. 
- Copy dependency jars ( for graphite ) from packaged opt to the flink lib.
- Configure graphite whitelist for reading flink monitor logs.

# Test details:
Verified for AWS:
UBUNTU - PICO -CDH & HDP
UBUNTU - STD -CDH & HDP
RHEL - PICO -CDH & HDP
RHEL - STD -CDH & HDP

### CODE REVIEW

**Review by trsmith2 — CHANGES_REQUESTED** [review]
If the code is sending metrics towards graphite then it should target a graphite role, not an edge node role

File: salt/flink/init.sls [hunk 1]
```diff
@@ -27,6 +27,17 @@
 {% set hadoop_home_bin = '/opt/cloudera/parcels/CDH/bin' %}
 {% endif %}
 
+# edge node IP
+{% set edge_nodes = salt['pnda.get_hosts_for_role']('hadoop_edge') %}
```
Comment (jeclarke) [inline salt/flink/init.sls]: Use the specific graphite role, rather than assuming it will be on the hadoop-edge node.

File: salt/flink/init.sls [hunk 2]
```diff
@@ -108,3 +121,7 @@ flink-history_server_start_service:
     - name: flink-history-server
     - enable: True
     - reload: True
+
+flink_metric_dep_jars-copy-to-flink-lib:
```
Comment (jeclarke) [inline salt/flink/init.sls]: Use the salt file.copy state rather than cmd.run.

**Review by jeclarke — CHANGES_REQUESTED** [review]

File: salt/flink/init.sls [hunk 3]
```diff
@@ -108,3 +114,12 @@ flink-history_server_start_service:
     - name: flink-history-server
     - enable: True
     - reload: True
+
+{% set dep_jars = salt['file.find']('%s/opt/' %flink_real_dir, name="*graphite*") %}
+{% for file in dep_jars %}
+{%- set filename = salt['file.basename'](file) %}
+copy_dependency_jar-{{ filename }}:
```
Comment (jgarnier) [inline salt/flink/init.sls]: could you prefix the state id with flink- please?

**Review by jgarnier — CHANGES_REQUESTED** [review]
add minor comment and could you rebase it with develop as there is currently some conflicts. thanks

**Pull request merged**